### PR TITLE
[ROCm][CI] Add MLA decode accuracy and determinism tests

### PR DESCRIPTION
**Pull request by aarushjain29**
Test the production BF16 decode path through rocm_aiter_ops.mla_decode_fwd which is the hot path for DeepSeek-V3/V4 inference on MI300/MI355.

Covers:
- Smoke test (shape, dtype, finite, non-zero)
- Accuracy vs PyTorch reference (absorbed MLA formulation)
- Parametrized sweep: nhead in {16, 128}, batch in {1, 4, 16}, seq in {16, 256}
- Bitwise determinism across 4 runs

Uses atol=0.01, rtol=0.0, pass rate >= 99.999% for BF16 attention. Requires MI3xx hardware for kernel execution.

## Purpose
Add kernel-level correctness tests for the AITER MLA decode path (`rocm_aiter_ops.mla_decode_fwd`), which is the hot path for DeepSeek-V3/V4 inference on MI300/MI355. Currently there are no tests that validate the numerical accuracy of this kernel against a reference implementation.
This complements the existing metadata tests (`test_rocm_aiter_mla_decode_metadata.py`) and sparse metadata sync tests (`test_rocm_aiter_mla_sparse_metadata_sync.py`) which test the builder/caching logic but never exercise the kernel output.

### CODE REVIEW

File: tests/kernels/attention/test_rocm_aiter_mla_decode.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+            inputs["o"] = torch.zeros_like(inputs["o"])
+            out = _run_decode(inputs)
+            max_diff = (out.float() - out_first.float()).abs().max()
+            assert torch.equal(out_first, out), (
+                f"Run {run} differs from run 0 (max diff = {max_diff.item():.2e})"
+            )
+
+
+if __name__ == "__main__":
+    pytest.main([__file__, "-v"])
```
Comment (AndreasKaratzas) [inline tests/kernels/attention/test_rocm_aiter_mla_decode.py]: I don't think there is any reason to add this at the end.

**Review by AndreasKaratzas — APPROVED** [review]
LGTM
